File: backend/app/schemas/movie_preference.py
"""Validation and response contracts for a user's movie list."""

from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field

from app.schemas.movie import MovieSummaryResponse


class MovieListStatus(str, Enum):
    completed = "completed"
    watching = "watching"
    plan_to_watch = "plan_to_watch"


class MoviePreferenceUpsertRequest(BaseModel):
    status: MovieListStatus
    rating: int | None = Field(default=None, ge=1, le=10)
    is_favorite: bool = False


class MoviePreferenceResponse(BaseModel):
    id: str
    user_id: str
    movie_id: int
    status: MovieListStatus
    rating: int | None
    is_favorite: bool
    movie: MovieSummaryResponse
    created_at: datetime
    updated_at: datetime


class MoviePreferenceListResponse(BaseModel):
    results: list[MoviePreferenceResponse]

File: frontend/app/types/moviePreference.ts
import type { MovieSummary } from './movie'

export type MovieListStatus = 'completed' | 'watching' | 'plan_to_watch'

export interface MoviePreference {
  id: string
  user_id: string
  movie_id: number
  status: MovieListStatus
  rating: number | null
  is_favorite: boolean
  movie: MovieSummary
  created_at: string
  updated_at: string
}

export interface MoviePreferenceListResponse {
  results: MoviePreference[]
}

export interface MoviePreferenceInput {
  status: MovieListStatus
  rating: number | null
  is_favorite: boolean
}

export const MOVIE_STATUS_LABELS: Record<MovieListStatus, string> = {
  completed: 'Completed',
  watching: 'Watching',
  plan_to_watch: 'Plan to Watch',
}
